runtime/nvqir/cutensornet/simulator_mps_register.cpp

```diff
@@ -84,6 +84,78 @@ class SimulatorMPS : public SimulatorTensorNetBase {
     return numQubits;
   }
 
+  static std::vector<std::complex<double>> generateXX(double theta) {
+    const auto halfTheta = theta / 2.;
+    const std::complex<double> cos = std::cos(halfTheta);
+    const std::complex<double> isin = {0., std::sin(halfTheta)};
+    // Row-major
+    return {cos, 0.,    0.,  -isin, 0.,    cos, -isin, 0.,
+            0.,  -isin, cos, 0.,    -isin, 0.,  0.,    cos};
+  };
+
+  static std::vector<std::complex<double>> generateYY(double theta) {
+    const auto halfTheta = theta / 2.;
+    const std::complex<double> cos = std::cos(halfTheta);
+    const std::complex<double> isin = {0., std::sin(halfTheta)};
+    // Row-major
+    return {cos, 0.,    0.,  isin, 0.,   cos, -isin, 0.,
+            0.,  -isin, cos, 0.,   isin, 0.,  0.,    cos};
+  };
+
+  static std::vector<std::complex<double>> generateZZ(double theta) {
+    const std::complex<double> itheta2 = {0., theta / 2.0};
+    const std::complex<double> exp_itheta2 = std::exp(itheta2);
+    const std::complex<double> exp_minus_itheta2 = std::exp(-1.0 * itheta2);
+    // Row-major
+    return {exp_minus_itheta2, 0., 0., 0., 0., exp_itheta2,      0., 0., 0., 0.,
+            exp_itheta2,       0., 0., 0., 0., exp_minus_itheta2};
+  };
+
+  virtual void applyExpPauli(double theta,
+                             const std::vector<std::size_t> &controls,
+                             const std::vector<std::size_t> &qubitIds,
+                             const cudaq::spin_op &op) override {
+    // Special handling for equivalence of Rxx(theta), Ryy(theta), Rzz(theta)
+    // expressed as exp_pauli.
+    //  Note: for MPS, the runtime is ~ linear with the number of 2-body gates
+    //  (gate split procedure).
+    // Hence, we check if this is a Rxx(theta), Ryy(theta), or Rzz(theta), which
+    // are commonly-used gates and apply the operation directly (the base
+    // decomposition will result in 2 CNOT gates).
+    const auto shouldHandlePauliOp =
```

Comment:
What would happen if `op = "XXXX"`? Is that the same as `"XXII"` followed by `"IIXX"`? If so, should we attempt to handle that by iterating over the elements in the `op` looking for neighboring identical Paulis, and just sending the remaining Paulis to the base class implementation?

Reply:
If `op = "XXXX"`, we don't handle it and let the base class performs the decomposition (with CNOT ladders before and after). It's not the same as `"XXII"` followed by `"IIXX"` though if users wrote the circuit in terms of `exp_pauli`, e.g.,
`exp_pauli(theta, 'XX', q0, q1)` followed by `exp_pauli(theta, 'XX', q2, q3)`.